src/App.jsx

```diff
@@ -1,41 +1,86 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { getPosts, getUserPosts } from './api/posts';
+import { getUsers } from './api/users';
+// import { post } from './api/api';
 
-const App = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App = () => {
+  const [posts, setPosts] = useState([]);
+  const [users, setUsers] = useState([]);
+  const [selectedPostId, setSelectedPostId] = useState(0);
+
+  useEffect(() => {
+    getPosts()
+      .then(data => setPosts(data));
+    getUsers()
+      .then(data => setUsers(data));
+  }, []);
+
+  const selectedUserPosts = ({ target }) => {
+    if (+target.value === 0) {
+      getPosts()
+        .then(data => setPosts(data));
+    }
+
+    getUserPosts(target.value)
+      .then(data => setPosts(data));
+  };
+
+  const setSelectPost = (postId) => {
+    if (postId === selectedPostId) {
```

Comment:
You could declare this function directly in the `PostsList` component and not pass it with props. It needed only there